api/v1/views/users.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python3
+"""Cities"""
+from flask import jsonify, request, abort
+from api.v1.views import app_views
+
+@app_views.route('/users', methods=['GET', 'POST'], strict_slashes=False)
+@app_views.route('/users/<user_id>', methods=['GET', 'DELETE', 'PUT'], strict_slashes=False)
+def api_users(user_id=None):
+    """Retrives, deletes, creates and updates a User object"""
+    from models import storage
+    from models.user import User
+    method = request.method
+
+    if method == 'GET':
+        if user_id is None:
+            users = storage.all(User).values
+            return jsonify([user.to_dict() for user in users])
+        else:
+            obj = storage.get(User, user_id)
+            return abort(404) if obj is None else jsonify(obj.to_dict())
+    elif method == 'DELETE':
+        if user_id is None or storage.get(User, user_id) is None:
+            abort(404)
+        else:
+            storage.delete(User, user_id)
```

Comment:
Delete takes an object, not a class and an id. You need to change this to send it an object (probably using the storage.get method)